File: StringToLinkedlist.c
//Write a program to convert a string into a linked list of characters.

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Node structure
struct Node {
    char data;
    struct Node* next;
};

// Function to create a new node
struct Node* createNode(char ch) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = ch;
    newNode->next = NULL;
    return newNode;
}

// Function to convert string to linked list
struct Node* stringToLinkedList(char* str) {
    struct Node *head = NULL, *tail = NULL;
    int i;

    for (i = 0; str[i] != '\0'; i++) {
        struct Node* newNode = createNode(str[i]);
        if (head == NULL) {
            head = tail = newNode;
        } else {
            tail->next = newNode;
            tail = newNode;
        }
    }
    return head;
}

// Function to display the linked list
void display(struct Node* head) {
    printf("Linked list of characters: ");
    while (head != NULL) {
        printf("%c -> ", head->data);
        head = head->next;
    }
    printf("NULL\n");
}

int main() {
    char str[100];
    printf("Enter a string: ");
    fgets(str, sizeof(str), stdin);

    // Remove newline character if present
    str[strcspn(str, "\n")] = '\0';

    struct Node* head = stringToLinkedList(str);
    display(head);

    return 0;
}

File: stringToLinkedlist.py
#Write a program to convert a string into a linked list of characters.

class CharNode:
    def __init__(self, char):
        self.data = char
        self.next = None

def string_to_linked_list(string):
    head = None
    tail = None
    for ch in string:
        new_node = CharNode(ch)
        if head is None:
            head = tail = new_node
        else:
            tail.next = new_node
            tail = new_node
    return head

def display(head):
    print("Linked list of characters:", end=" ")
    while head:
        print(f"{head.data} -> ", end="")
        head = head.next
    print("NULL")

# Example usage
s = input("Enter a string: ")
head = string_to_linked_list(s)
display(head)
